conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/EndpointNameHeaderEnrichmentContract.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.jaxrs.feignimpl;
+
+import feign.Contract;
+import feign.MethodMetadata;
+import java.lang.reflect.Method;
+
+/**
+ * Contract to capture the endpoint name (method name) and pass it to a feign client.
+ *
+ * This should be considered internal API and should not be depended upon.
+ */
+public final class EndpointNameHeaderEnrichmentContract extends AbstractDelegatingContract {
```

Comment:
out of curiosity, why does this have to be public? could we just pop it in the same package as the JaxRsClient class?

Reply:
AbstractDelegatingContract Is package private, we could duplicate it or make it public instead.

Reply:
Eh, no strong preference tbh